tests/testthat/test-trans-numeric.R

```diff
@@ -116,3 +116,8 @@ test_that("Yeo-Johnson transform works", {
   expect_equal(yj_trans(lambdas[2])$transform(x[[2]]), expected_data[[2]])
   expect_equal(yj_trans(lambdas[3])$transform(x[[3]]), expected_data[[3]])
 })
+
+test_that("probability transforms have domain (0,1)", {
+  expect_true(all(logit_trans()$domain == c(0, 1)))
+  expect_true(all(probit_trans()$domain == c(0, 1)))
```

Comment:
I think this is not an accurate test.
A better one could be 
```r
  expect_true(all(logit_trans()$domain %in% c(0, 1)))
  expect_true(all(probit_trans()$domain %in% c(0, 1)))
```

Or if you always expect a length 2, exactly `c(0, 1)`
```r
  expect_equal(logit_trans()$domain, c(0, 1))
```

Or if you are not sure of the order

```r
  expect_setequal(logit_trans()$domain, c(0, 1))
```

Reply:
Isn't this testing that the domain is exactly `c(0, 1)`? Then it should be `expect_equal(logit_trans()$domain, c(0, 1))` or `expect_identical(logit_trans()$domain, c(0, 1))`

Reply:
I think you are right. 
I wasn't exactly sure, but I knew that  `expect_true(all(logit_trans()$domain == c(0, 1)))`

Reply:
e.g.
```r
c(1, 1) == c(2, 1)
```
...yields
```r
FALSE TRUE
```
...a logical vector, which when tested with `all()` would yield `FALSE` (in this case). It is equivalent to `expect_equal`.

So not sure what "not an accurate test" means? I agree that the `expect_equal` syntax is preferrable, so will make that change.

I think the length test would be more appropriate as a general test of all transform generators.

Reply:
It is just that `all()` doesn't handle edge cases very well.
```r
all(integer(0) == c(0, 1)) |> testthat::expect_true()
```

This test passes.

Reply:
to be clear, the explicit intent here is that the domain values lower == 0 & upper == 1, which is equivalent to all(c(lower, upper) == c(0, 1)). this isn't an %in% relationship

Reply:
Ok, the correct test would therefore be 

```r
expect_equal(logit_trans()$domain, c(0, 1))
```

Because in the (unlikely) scenario that
`logit_trans()$domain = integer(0)`, the current test would pass even though it is incorrect.

Reply:
The key defect in the all approach seems to be the "empty set" case, so that aspect of it actually does invalidate the test.

Reply:
also: just now seeing replies. not a crazy person ignoring your points.

Reply:
No problem. Sorry. I wanted to make sure I was clear because I saw this exact problem in another repo that hid a bug silently for a while. :) 

```r
  expect_equal(logit_trans()$domain, c(0, 1))
```
note the `,` instead of `==`